pkg/controllers/util/annotation/annotation.go

```diff
@@ -67,7 +58,7 @@ func HasAnnotationKeyValue(obj runtime.Object, key, value string) (bool, error)
 // AddAnnotation adds the given annotation key and value to the given objects ObjectMeta,
 // and overwrites the annotation value if it already exists.
 // Returns true if the object was updated.
-func AddAnnotation(obj runtime.Object, key, value string) (bool, error) {
+func AddAnnotation(obj metav1.Object, key, value string) (bool, error) {
```

Comment:
Besides avoiding the `meta.Accessor` overhead, another point of this refactor is to allow calling `AddAnnotation` without having to handle the returned error (given the current code paths, the old implementation of `AddAnnotation` would never return an error anyway).

The following is a possible implementation of `AddAnnotation` that satisfies the requirements:

```golang
func AddAnnotation(obj metav1.Object, key, value string) (bool) {
	has := HasAnnotationKeyValue(obj, key, value)
	if has {
		return false
	}
	annotations := obj.GetAnnotations()
	if annotations == nil {
		annotations = make(map[string]string)
	}
	annotations[key] = value
	obj.SetAnnotations(annotations)
	return true
}
```

Naturally, since the return signature has changed, all the call sites need to be changed (which is a good thing because the old error handling code is merely unnecessary bloat).